### ISSUE TITLE
[Request]: Add YimMenu User Detection.

### ISSUE THREAD

**Issue by Adler-D3R** [Yimura/YimMenu#727]
### Problem

It would be nice if every YimMenu user could know if another player uses it as well.

### Solution

A solution would be sending out a specific SE when entering a session, and listening to it when received. This SE can be anything, even a not used one. When received, add the sender to a list, which is the YimMenu User List, and here you go.

### Reason

I think that would be great to encourage players to stay friendly toward others.

**Comment by sch-lda** [comment]
I think ... other menus and rockstar can also detect Yimmenu easily by this way because Yimmenu is an open source project

**Comment by Yimura** [comment]
Script Events are super easy to detected, we can go lower in the net stack and make a custom netMsg.

**Comment by maybegreat48** [comment]
TLDR: Doing that will just lead to YimMenu being added to code such as this
```c++
   if ( *&v12 > 0x1B0EF412Ci64 )
    {
      if ( *&v12 > 0x113252DDCCi64 )
      {
        if ( *&v12 == 0x1161FBF626i64 )
        {
          v13 = "Hypno / Cobra";
          goto LABEL_37;
        }
        if ( *&v12 == 0x2625F41F20E34i64 )
        {
          v13 = "Cozy";
          goto LABEL_37;
        }
      }
      else
      {
        switch ( *&v12 )
        {
          case 0x113252DDCCi64:
            v13 = "X-Force";
            goto LABEL_37;
          case 0x1B0EF454Ai64:
            v13 = "Terror";
            goto LABEL_37;
          case 0x1F2971AE6i64:
            v13 = "DarkCheats";
            goto LABEL_37;
          case 0x10E93E1A2Fi64:
            v13 = "Mister//ModzZ PC";
            goto LABEL_37;
        }
      }
    }
    else
    {
      if ( *&v12 == 0x1B0EF412Ci64 )
      {
        v13 = "Sodium";
        goto LABEL_37;
      }
      if ( *&v12 > 7263263744i64 )
      {
        switch ( *&v12 )
        {
          case 0x1B0EC9C05i64:
            v13 = "Volcano VIP";
            goto LABEL_37;
          case 0x1B0EC9C14i64:
            v13 = "Position";
            goto LABEL_37;
          case 0x1B0ECAEDDi64:
            v13 = "Menace";
            goto LABEL_37;
        }
      }
      else
      {
        if ( *&v12 == 7263263744i64 || *&v12 == -1890951223i64 )
        {
          v13 = "Explosive";
          goto LABEL_37;
        }
        if ( *&v12 == -123536367i64 || *&v12 == -2289045i64 )
        {
          v13 = "Monopol";
LABEL_37:
          menuDetected(a1, v13);
```
Doesn't matter how you decide to transmit the signal, other menus will add it to their detection list and players can simply block join/kick all YimMenu users if they wanted to. Most P2Cs deal with this by using an API, but it can be easily tampered with and could cause the menu to not be allowed in UnknownCheats.


```
I think that would be great to encourage players to stay friendly toward others.
```
I HIGHLY doubt it

**Comment by Yimura** [comment]
Yeah remaining undetected is less likely to get you targeted than being identifiable. I don't think this should become a thing in YimMenu, with how YimMenu is opensource this will be impossible to sneakily add.

Adding to it, using an API would mean we expose all YimMenu users publicly.

**Comment by ghost** [comment]
> 

Almost all menus on here are skidded from the same source, dang.

**Comment by Vali0004** [comment]
> TLDR: Doing that will just lead to YimMenu being added to code such as this
> 
> ```c++
>    if ( *&v12 > 0x1B0EF412Ci64 )
>     {
>       if ( *&v12 > 0x113252DDCCi64 )
>       {
>         if ( *&v12 == 0x1161FBF626i64 )
>         {
>           v13 = "Hypno / Cobra";
>           goto LABEL_37;
>         }
>         if ( *&v12 == 0x2625F41F20E34i64 )
>         {
>           v13 = "Cozy";
>           goto LABEL_37;
>         }
>       }
>       else
>       {
>         switch ( *&v12 )
>         {
>           case 0x113252DDCCi64:
>             v13 = "X-Force";
>             goto LABEL_37;
>           case 0x1B0EF454Ai64:
>             v13 = "Terror";
>             goto LABEL_37;
>           case 0x1F2971AE6i64:
>             v13 = "DarkCheats";
>             goto LABEL_37;
>           case 0x10E93E1A2Fi64:
>             v13 = "Mister//ModzZ PC";
>             goto LABEL_37;
>         }
>       }
>     }
>     else
>     {
>       if ( *&v12 == 0x1B0EF412Ci64 )
>       {
>         v13 = "Sodium";
>         goto LABEL_37;
>       }
>       if ( *&v12 > 7263263744i64 )
>       {
>         switch ( *&v12 )
>         {
>           case 0x1B0EC9C05i64:
>             v13 = "Volcano VIP";
>             goto LABEL_37;
>           case 0x1B0EC9C14i64:
>             v13 = "Position";
>             goto LABEL_37;
>           case 0x1B0ECAEDDi64:
>             v13 = "Menace";
>             goto LABEL_37;
>         }
>       }
>       else
>       {
>         if ( *&v12 == 7263263744i64 || *&v12 == -1890951223i64 )
>         {
>           v13 = "Explosive";
>           goto LABEL_37;
>         }
>         if ( *&v12 == -123536367i64 || *&v12 == -2289045i64 )
>         {
>           v13 = "Monopol";
> LABEL_37:
>           menuDetected(a1, v13);
> ```
> 
> Doesn't matter how you decide to transmit the signal, other menus will add it to their detection list and players can simply block join/kick all YimMenu users if they wanted to. Most P2Cs deal with this by using an API, but it can be easily tampered with and could cause the menu to not be allowed in UnknownCheats.
> 
> ```
> I think that would be great to encourage players to stay friendly toward others.
> ```
> 
> I HIGHLY doubt it

My old garbage menu is in that list. God that was ages ago, totally forgot about it.

**Comment by izeerF** [comment]
it can be done with some cheap/free http server, but do anyone really need that?

**Comment by Yimura** [comment]
> it can be done with some cheap/free http server, but do anyone really need that?

Hosting is not the problem, my problem is that it would expose those users that opt-into it. YimMenu does not have any authentication service built-in so the way to update and fetch such data would also be exposed meaning that any other menu out there could use it to identify YimMenu users as well.

**Comment by izeerF** [comment]
> > it can be done with some cheap/free http server, but do anyone really need that?
> 
> Hosting is not the problem, my problem is that it would expose those users that opt-into it. YimMenu does not have any authentication service built-in so the way to update and fetch such data would also be exposed meaning that any other menu out there could use it to identify YimMenu users as well.

but if it would be done by external process?

**Comment by Yimura** [comment]
> but if it would be done by external process?

There are two ways of doing this and both expose YimMenu users to other modders.

1. Internally, by using the existing network structure of the game to send data in the session to identify yourself as a YimMenu user. This can be easily picked up by other modders if they were to implement the same into their projects.
2. Externally, by using a web server with an API tied to it. This would require some extra work to get working but can still be abused by anyone as the endpoint would have no authentication on it. (YimMenu doesn't require you to authenticate/login after all)

First one would immediately expose you in a session to all modders that would bother processing the identification but would be safer than the external one as that one would allow you to export all YimMenu users that make use of the feature by iterating over all the RID's in existence (it's an identifier that can be iterated after all)

Obviously we can mitigate this somewhat by adding ratelimits to the endpoints but all a menu has to do is request it for all the users in a session once the enter it and it would still expose YimMenu users in ways that aren't intended.

Because of this I myself would not want to add this to YimMenu and is something that could only work for a short while for closed source menu's but even those would end up being exposed by people reverse engineering the binary.

For those reason we will not pursue this request further and will deny any PR's attempting to add it.

**Comment by izeerF** [comment]
> 2\. This would require some extra work to get working but can still be abused by anyone as the endpoint would have no authentication on it

ye true, i forgore that it would need some auth like payed menus
